src/pages/beneficiaries/Beneficiaries.js

```diff
@@ -12,71 +15,228 @@ const FILTER_MAP = {
+    const [activeFilter, setActiveFilter] = useState(true);
+    const [archivedFilter, setArchivedFilter] = useState(false);
+    const [interestsFilter, setInterestsFilter] = useState(false);
+    const [computers, setComputers] = useState(false);
+    const [spokenEnglish, setSpokenEnglish] = useState(false);
+    const [literacy, setLiteracy] = useState(false);
+    const [math, setMath] = useState(false);
+    const [bengali, setBengali] = useState(false);
+    const [arts, setArts] = useState(false);
+    const [bakery, setBakery] = useState(false);
+    const [counseling, setCounseling] = useState(false);
+    const [numInterests, setNumInterests] = useState(0);
 
     const deleteBfc = (id) => {
         console.log(id);
-        fetch(`http://localhost:3000/beneficiaries?beneficiaryID=${id}`, {
+        fetch(`http://localhost:3000/beneficiaries?id=${id}`, {
             method: "DELETE",
         }).then(async () => {
             try {
                 let data = await fetch("http://localhost:3000/beneficiaries");
                 data = await data.json();
-                setBeneficiary(data);
-                console.log(beneficiary);
+                setBeneficiaries(data);
+                console.log("beneficiaries: " + beneficiaries);
             } catch (error) {
                 console.error(error);
             }
         });
     };
 
-    const sortByName = () => {
-        let data = [...beneficiary];
+    const sortByFirstName = () => {
+        let data = [...beneficiaries];
         data.sort((a, b) => a.firstName.localeCompare(b.firstName));
-        setBeneficiary(data);
+        setBeneficiaries(data);
+    };
+
+    const sortBackwardsByFirstName = () => {
+        let data = [...beneficiaries];
+        data.sort((a, b) => b.firstName.localeCompare(a.firstName));
+        setBeneficiaries(data);
+    };
+
+    const sortByLastName = () => {
+        let data = [...beneficiaries];
+        data.sort((a, b) => a.lastName.localeCompare(b.lastName));
+        setBeneficiaries(data);
+    };
+
+    const sortBackwardsByLastName = () => {
+        let data = [...beneficiaries];
+        data.sort((a, b) => b.lastName.localeCompare(a.lastName));
+        setBeneficiaries(data);
     };
 
     const sortByDate = () => {
-        let data = [...beneficiary];
+        let data = [...beneficiaries];
         data.sort(function (a, b) {
             const dateA = new Date(a.joinDate);
             const dateB = new Date(b.joinDate);
             return dateB - dateA;
         });
-        console.log(data);
-        setBeneficiary(data);
+        setBeneficiaries(data);
+    };
+
+    const sortBackwardsByDate = () => {
+        let data = [...beneficiaries];
+        data.sort(function (a, b) {
+            const dateA = new Date(a.joinDate);
+            const dateB = new Date(b.joinDate);
+            return dateA - dateB;
+        });
+        setBeneficiaries(data);
+    };
```

Comment:
These sort functions seem to be very similar. Try optimizing these in the future.